introspection.go

```diff
@@ -16,12 +16,20 @@ type Service struct {
 	SchemaSource string
 	Schema       *ast.Schema
 	Status       string
+
+	client *GraphQLClient
 }
 
-func NewService(serviceURL string) *Service {
-	return &Service{
+func NewService(serviceURL string, client *GraphQLClient) *Service {
```

Comment:
I'm wondering if we should maybe do the client initialization here in `NewService`, this way we could have `NewClient(WithUserAgent(generateBrambleUserAgent("update")))` in only one place.
We'll lose the ability to inject a different client but I don't think we currently need that.

What do you think?

Reply:
Yep that sounds good to me. I wasn't sure how reusable service needed to be so didn't want to prescribe only one user agent but if it's ok I'll switch it.